airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/utils/airbyte_entrypoint.py

```diff
@@ -0,0 +1,132 @@
+#
+# Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+#
+
+import docker
+import json
+import logging
+import pytest
+import shutil
+import sys
+import tempfile
+from dataclasses import dataclass
+from docker.client import DockerClient
+from docker.models.images import Image
+from pathlib import Path
+from typing import List
+
+from .connector_runner import ConnectorRunner
+from .input_config import InputConfig
+
+logger = logging.getLogger(__name__)
+
+
+def reset_logging():
+    """Removes all previous default loggers"""
+    for handler in logging.root.handlers[:]:
+        logging.root.removeHandler(handler)
+    logging.basicConfig(
+        level=logging.INFO,
+        format='%(levelname)s # %(message)s'
+    )
+    sys.excepthook = None
+
+
+@dataclass
+class AirbyteEntrypoint:
+    """Base launch class"""
+
+    config: InputConfig
+
+    def _run_pytest(self, args: List[str]) -> int:
+        """"try to run the tool pytest into source folder"""
+        args += "-r fEsx --capture=no -vv --log-level=WARNING --color=yes --force-sugar".split(" ")
+        return pytest.main(args)
+
+    def run_integration_tests(self) -> int:
+        """"try to run integration tests into source folder"""
+        return self._run_pytest([
+            str(self.config.source_dir / "integration_tests"),
+            "-p", "source_acceptance_test.plugin",
+            "--acceptance-test-config", str(self.config.source_dir)
+        ])
+
+    def __run_in_docker(self) -> int:
+        """"try to run tests into a new docker container"""
+        volumes = {
+            "/var/run/docker.sock": {'bind': '/var/run/docker.sock', 'mode': 'rw'},
+            "/tmp": {'bind': '/tmp', 'mode': 'rw'},
+        }
+
+        client = docker.from_env()
+        if self.config.is_python:
+            image = self.__build_py_test_image(client)
+        else:
+            raise Exception("don't support other language...")
+        cmd = " ".join(sys.argv[1:]).replace("--acceptance-tests", "")
+        working_dir = image.attrs["Config"]["WorkingDir"] or "/airbyte/integration_code"
+        cmd = cmd.replace(str(self.config.source_dir), working_dir)
+
+        container = client.containers.run(
+            image=image, detach=True,
+            command=cmd,
+            volumes=volumes, auto_remove=True
+        )
+        for line in ConnectorRunner.read(container, with_ext=False):
+            print(line)
+        exit_status = container.wait()
+        logger.info(f"Result: {exit_status}")
+        return exit_status["StatusCode"]
+
+    def __build_py_test_image(self, client: DockerClient) -> Image:
```

Comment:
when I look at this function I can't get rid of the feeling that Python is not the right tool for the job